### PR TITLE
Add Optional base_url Parameter to Clients

### PR DESCRIPTION
**Pull request by yunusemreozdemir**
This PR adds the optional `base_url` parameter to QStash clients. This can be used to change the url of the QStash client for development/testing purposes.

### CODE REVIEW

File: qstash/client.py [hunk 1]
```diff
@@ -19,12 +19,17 @@ def __init__(
...
     ) -> None:
         """
         :param token: The authorization token from the Upstash console.
         :param retry: Configures how the client should retry requests.
         """
-        http = HttpClient(token, retry)
+        http = HttpClient(
+            token,
+            retry,
+            base_url=(base_url or "https://qstash.upstash.io"),
```
Comment (CahidArda) [inline qstash/client.py:31]: no need for another default here

File: qstash/asyncio/client.py [hunk 2]
```diff
@@ -18,12 +18,17 @@ def __init__(
...
     ) -> None:
         """
         :param token: The authorization token from the Upstash console.
         :param retry: Configures how the client should retry requests.
         """
-        http = AsyncHttpClient(token, retry)
+        http = AsyncHttpClient(
+            token,
+            retry,
+            base_url=(base_url or "https://qstash.upstash.io"),
```
Comment (CahidArda) [inline qstash/asyncio/client.py:30]: no need for another default here

File: qstash/asyncio/http.py [hunk 3]
```diff
@@ -19,6 +19,7 @@ def __init__(
         self,
         token: str,
         retry: Optional[Union[Literal[False], RetryConfig]],
+        base_url: str,
```
Comment (CahidArda) [inline qstash/asyncio/http.py:22]: can be optional with default value None

File: qstash/http.py [hunk 4]
```diff
@@ -102,6 +102,7 @@ def __init__(
         self,
         token: str,
         retry: Optional[Union[Literal[False], RetryConfig]],
+        base_url: str,
```
Comment (CahidArda) [inline qstash/http.py:105]: can be optional with default value none

File: qstash/asyncio/http.py [hunk 5]
```diff
@@ -33,6 +34,8 @@ def __init__(
             timeout=DEFAULT_TIMEOUT,
         )
 
+        self._base_url = base_url
```
Comment (mdumandag) [inline qstash/asyncio/http.py:37]: you can make this `base_url.rstrip("/") if base_url else BASE_URL` to get rid of tri-operand `or` in every call to request, and restore the rstrip behavior

**Review by CahidArda — APPROVED** [review]

**Pull request merged**